functions/go/apply-replacements/replacements/replacements.go

```diff
@@ -0,0 +1,71 @@
+package replacements
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sigs.k8s.io/kustomize/api/filters/replacement"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const fnConfigKind = "ApplyReplacements"
+const fnConfigApiVersion = "fn.kpt.dev/v1alpha1"
+
+func ApplyReplacements(rl *fn.ResourceList) error {
+	r := Replacements{}
+	return r.Process(rl)
+}
+
+type Replacements struct {
+	Replacements []types.Replacement `json:"replacements,omitempty" yaml:"replacements,omitempty"`
+}
+
+// Config initializes Replacements from a functionConfig fn.KubeObject
+func (r *Replacements) Config(functionConfig *fn.KubeObject) error {
+	if functionConfig.Kind() != fnConfigKind || functionConfig.APIVersion() != fnConfigApiVersion {
+		return fmt.Errorf("received functionConfig of kind %s and apiVersion %s, " +
+			"only functionConfig of kind %s and apiVersion %s is supported",
+			functionConfig.Kind(), functionConfig.APIVersion(), fnConfigKind, fnConfigApiVersion)
+	}
+	r.Replacements = []types.Replacement{}
+	if err := functionConfig.As(r); err != nil {
+		return fmt.Errorf("unable to convert functionConfig to %s:\n%w",
+			"replacements", err)
+	}
+	return nil
+}
+
+// Process configures the replacements and transformers them.
+func (r *Replacements) Process(rl *fn.ResourceList) error {
+	if err := r.Config(rl.FunctionConfig); err != nil {
+		return err
+	}
+	transformedItems, err := r.Transform(rl.Items)
+	if err != nil {
+		return err
+	}
+	rl.Items = transformedItems
+	return nil
+}
+
+// Transform runs the replacement filter in order to apply the replacements - this
+// does the actual work.
+func (r *Replacements) Transform(items []*fn.KubeObject) ([]*fn.KubeObject, error) {
+	var transformedItems []*fn.KubeObject
+	var nodes []*yaml.RNode
+	for _, obj := range items {
+		objRN := obj.ToRNode()
+		nodes = append(nodes, objRN)
+	}
+	transformedNodes, err := replacement.Filter{
+		Replacements:  r.Replacements,
+	}.Filter(nodes)
```

Comment:
In lieu of supporting the conversion between RNode to KubeObjects, we shall have replacement.Filter (maybe forked to sdk) to accept KubeObjects intead. WDYT?

Reply:
I have a similar problem with starlark fn: https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/starlark/third_party/sigs.k8s.io/kustomize/kyaml/fn/runtime/starlark/starlark.go#L51
The `Filter` takes a list of RNode as input. I will need to convert KubeObjects to RNodes.

Reply:
Good to now, I'll try prioritizing this fork work.

Reply:
@yuwenma did you have some concerns with @droot's suggestion in standup? If the sdk can convert a yaml string to KubeObject, than we can do RNode->yaml->KubeObject easily. Do you prefer to fork instead?

Reply:
"convert a yaml string to KubeObject" --> yes, we have `fn.ParseKubeObject()` that converts a byte slice to KubeObject. From the long term perspective considering the code quality and maintenance cost, forking is more clean and reliable than RNode->yaml-> KubeObject.  I would say `fn.ParseKubeObject()` is a temporary workaround for replacement transformer.

Reply:
Ah, makes sense. Wasn't aware of fn.ParseKubeObject(). I will use that for now and will use the fork when it is ready. Thank you!